Snakefile

```diff
@@ -8,6 +8,82 @@ wildcard_constraints:
+# Allow this to work from a separate workdir by using a config in that workdir
+# which extends one of our base configs
+# ----------------------------------------------------------------------------
+if os.path.exists("config.yaml"):
+    configfile: "config.yaml"
+    # See commentary below
+    # print("This doesn't work as expected! See commentary in Snakefile", file=sys.stderr)
+    # exit(2)
+
+if config.get('extends', False):
+    extend_path = os.path.join(workflow.basedir, "config", config['extends'])
+    if not os.path.isfile(extend_path):
+        sys.exit(f"Your config tried to extend {config['extends']!r} but this doesn't exist. It must be relative to {os.path.join(workflow.basedir, 'config')}")
+    configfile: extend_path
+
+# NOTE:
+# In the situation where we're running outside of the repo, and we have a custom config YAML
+# such as `foo.yaml`:
+#        extends: h5n1-cattle-outbreak.yaml
+#        segments: ['pb2']
+# If we run with `--configfile foo.yaml` then the merging behaviour is strange (to me!)
+# We've clearly parsed the --configfile, as we have config['extends']="h5n1-cattle-outbreak.yaml",
+# and we do merge in all the config values of `h5n1-cattle-outbreak.yaml` (via the above code)
+# so I expected we'd therefore have config['segments']=['genome', 'pb2', 'pb1', ...]
+# as defined in 'h5n1-cattle-outbreak.yaml', however we end up with only config['segments']=['pb2'].
+# So it seems like the `--configfile` definitions are being re-applied a second time?!?
+#
+# This _is not the case_ when we use the `os.path.exists("config.yaml")` approach,
+# which is why it's not going to work without the following additional update_config
+# step (or something else?)
+
+if os.path.exists("config.yaml"):
+    # Following <https://github.com/snakemake/snakemake/blob/76d53290a003891c5ee41f81e8eb4821c406255d/snakemake/common/configfile.py#L7-L33>
+    import yte
+    with open("config.yaml", encoding='utf-8') as f:
+        overwrite_config = yte.process_yaml(f, require_use_yte=True)
+    snakemake.utils.update_config(config, overwrite_config)
+
+from pprint import pp; pp(config, stream=sys.stderr) # TODO XXX remove
+
+
+def resolve_config_path(original_path, wildcards=None):
+    """
+    Resolve a relative *path* given in a configuration value.
+    Resolves *path* as relative to the workflow's ``config/`` directory (i.e.
+    ``os.path.join(workflow.basedir, "config", path)``) if it doesn't exist
+    in the workflow's analysis directory (i.e. the current working
+    directory, or workdir, usually given by ``--directory`` (``-d``)).
+    This behaviour allows a default configuration value to point to a default
+    auxiliary file while also letting the file used be overridden either by
+    setting an alternate file path in the configuration or by creating a file
+    with the conventional name in the workflow's analysis directory.
+    """
+    path = original_path.format(**wildcards) if wildcards else original_path
```

Comment:
Hmm. I've shifted to `expand` but I see it the other way round - if `expand` API changes (!) we now have to change this function, whereas with `.format` it'll continue to work as expected.

Reply:
I guess my thinking is that if the `expand()` API changes this will be the least of our worries. ;-)

My point is: we're doing wildcard expansion, so we should use Snakemake's facilities for it.

Reply:
Agree to disagree. I think `format` does exactly what we want here, whereas `expand` can be made to do what we want. You're leading this work so I consider this resolved.